ssl-impl/src/main/java/io/crate/protocols/postgres/ssl/SslConfiguration.java

```diff
@@ -0,0 +1,248 @@
+        }
+    }
+
+    private static SslContext buildSSLServerContext(final PrivateKey privateKey,
+                                                    final X509Certificate[] cert,
+                                                    final X509Certificate[] trustedCerts,
+                                                    final Iterable<String> ciphers,
+                                                    final SslProvider sslProvider) throws SSLException {
+        final SslContextBuilder sslContextBuilder =
+            SslContextBuilder
+                .forServer(privateKey, cert)
+                .ciphers(ciphers)
+                .applicationProtocolConfig(ApplicationProtocolConfig.DISABLED)
+                .clientAuth(ClientAuth.OPTIONAL)
+                .sessionCacheSize(0)
+                .sessionTimeout(0)
+                .sslProvider(sslProvider);
+
+        if(trustedCerts != null && trustedCerts.length > 0) {
+            sslContextBuilder.trustManager(trustedCerts);
+        }
+
+        return buildSSLContext(sslContextBuilder);
+    }
+
+    private static SslContext buildSSLContext(final SslContextBuilder sslContextBuilder) throws SSLException {
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        SslContext sslContext;
+        try {
+            sslContext = AccessController.doPrivileged(
+                (PrivilegedExceptionAction<SslContext>) sslContextBuilder::build);
+        } catch (final PrivilegedActionException e) {
+            throw (SSLException) e.getCause();
+        }
+
+        return sslContext;
+    }
+
+    @VisibleForTesting
+    static TrustStoreSettings loadTrustStore(Settings settings) throws KeyStoreException,
+                                                                       IOException,
+                                                                       NoSuchAlgorithmException,
+                                                                       CertificateException {
+        String trustStorePath = SslConfigSettings.SSL_TRUSTSTORE_FILEPATH.setting().get(settings);
+        checkStorePath(trustStorePath, StoreType.trust);
+        String trustStorePassword = SslConfigSettings.SSL_TRUSTSTORE_PASSWORD.setting().get(settings);
+
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+
+        // load the store
+        trustStore.load(new FileInputStream(new File(trustStorePath)),
+                        trustStorePassword.isEmpty() ? null : trustStorePassword.toCharArray());
+
+        // initialize a trust manager factory with the trusted store
+        TrustManagerFactory trustFactory =
+            TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustFactory.init(trustStore);
+
+        // get the trust managers from the factory
```

Comment:
Instead of these kind of obvious comments I think some more high-level explanation of what's going on would be more helpful  =)

Some method and class level docs would be great